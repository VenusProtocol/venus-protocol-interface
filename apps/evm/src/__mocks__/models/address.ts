const address = '0x3d759121234cd36F8124C21aFe1c6852d2bEd848';

export const altAddress = '0xa258a693A403b7e98fd05EE9e1558C760308cFC7';

export default address;
