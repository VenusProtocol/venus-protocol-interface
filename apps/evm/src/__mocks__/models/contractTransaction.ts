import { ChainId } from 'types';
import type { Transaction } from 'viem';

const contractTransaction: Transaction = {
  hash: '0xfake-transaction-hash',
  type: 'legacy',
  from: '0x1d759121234cd36F8124C21aFe1c6852d2bEd848',
  to: '0x2d759121234cd36F8124C21aFe1c6852d2bEd834',
  nonce: 10,
  value: 0n,
  chainId: ChainId.BSC_TESTNET,
  blockHash: '0xfake-block-hash',
  blockNumber: 1123213n,
  gas: 12n,
  gasPrice: 1n,
  input: '0xfake-input',
  r: '0xfake-r',
  s: '0xfake-s',
  v: 3n,
  transactionIndex: 1,
  typeHex: '0xfake-type-hex',
};

export default contractTransaction;
