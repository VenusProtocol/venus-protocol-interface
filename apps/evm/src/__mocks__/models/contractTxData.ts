import type { ContractTxData } from 'types';

const contractTxData: ContractTxData<any, any> = {
  contract: {},
  methodName: 'fakeMethod',
  args: [],
};

export default contractTxData;
