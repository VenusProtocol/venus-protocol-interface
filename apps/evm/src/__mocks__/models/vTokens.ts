import type { VToken } from 'types';

import { bnb, busd, eth, lisUsd, luna, usdc, usdt, ust, wbnb, weth, xvs } from './tokens';

export const vXvs: VToken = {
  address: '0x6d6F697e34145Bb95c54E77482d97cc261Dc237E',
  decimals: 8,
  symbol: 'vXVS',
  underlyingToken: xvs,
};

export const vBnb: VToken = {
  address: '0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c',
  decimals: 8,
  symbol: 'vBNB',
  underlyingToken: bnb,
};

export const vUsdc: VToken = {
  address: '0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7',
  decimals: 8,
  symbol: 'vUSDC',
  underlyingToken: usdc,
};

export const vUsdtCorePool: VToken = {
  address: '0xb7526572FFE56AB9D7489838Bf2E18e3323b441A',
  decimals: 8,
  symbol: 'vUSDT',
  underlyingToken: usdt,
};

export const vUsdtDeFi: VToken = {
  address: '0x80CC30811e362aC9aB857C3d7875CbcCc0b65750',
  decimals: 8,
  symbol: 'vUSDT_DeFi',
  underlyingToken: usdt,
};

export const vUsdtGameFi: VToken = {
  address: '0x0bFE4e0B8A2a096A27e5B18b078d25be57C08634',
  decimals: 8,
  symbol: 'vUSDT_GameFi',
  underlyingToken: usdt,
};

export const vUsdtLiquidStakedBnb: VToken = {
  address: '0x2197d02cC9cd1ad51317A0a85A656a0c82383A7c',
  decimals: 8,
  symbol: 'vUSDT_Liquid_Staked_BNB',
  underlyingToken: usdt,
};

export const vUsdtMeme: VToken = {
  address: '0x3AF2bE7AbEF0f840b196D99d79F4B803a5dB14a1',
  decimals: 8,
  symbol: 'vUSDT_Meme',
  underlyingToken: usdt,
};

export const vUsdtStablecoins: VToken = {
  address: '0x3338988d0beb4419Acb8fE624218754053362D06',
  decimals: 8,
  symbol: 'vUSDT_Stablecoins',
  underlyingToken: usdt,
};

export const vUsdtTron: VToken = {
  address: '0x712774CBFFCBD60e9825871CcEFF2F917442b2c3',
  decimals: 8,
  symbol: 'vUSDT_Tron',
  underlyingToken: usdt,
};

export const vBusdCorePool: VToken = {
  address: '0x08e0A5575De71037aE36AbfAfb516595fE68e5e4',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vBusdStablecoins: VToken = {
  address: '0x8c8A1a0b6e1cb8058037F7bF24de6b79Aca5B7B0',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vUst: VToken = {
  address: '0xF206af85BC2761c4F876d27Bd474681CfB335EfA',
  decimals: 8,
  symbol: 'vUST',
  underlyingToken: ust,
};

export const vLuna: VToken = {
  address: '0x9C3015191d39cF1930F92EB7e7BCbd020bCA286a',
  decimals: 8,
  symbol: 'vLUNA',
  underlyingToken: luna,
};

export const vWeth: VToken = {
  address: '0xc2931B1fEa69b6D6dA65a50363A8D75d285e4da9',
  decimals: 8,
  symbol: 'vWETH',
  underlyingToken: weth,
};

export const vEth: VToken = {
  address: '0x86f8DfB7CA84455174EE9C3edd94867b51Da46BD',
  decimals: 8,
  symbol: 'vETH',
  underlyingToken: eth,
};

export const vLisUSD: VToken = {
  address: '0x170d3b2da05cc2124334240fB34ad1359e34C562',
  decimals: 8,
  symbol: 'vlisUSD',
  underlyingToken: lisUsd,
};

export const vWbnb: VToken = {
  address: '0x231dED0Dfc99634e52EE1a1329586bc970d773b3',
  decimals: 8,
  symbol: 'vWBNB',
  underlyingToken: wbnb,
};

export default [
  vXvs,
  vBnb,
  vUsdc,
  vUsdtCorePool,
  vUsdtDeFi,
  vUsdtGameFi,
  vUsdtLiquidStakedBnb,
  vUsdtMeme,
  vUsdtStablecoins,
  vUsdtTron,
  vBusdCorePool,
  vBusdStablecoins,
  vUst,
  vLuna,
  vEth,
  vLisUSD,
  vWbnb,
];
