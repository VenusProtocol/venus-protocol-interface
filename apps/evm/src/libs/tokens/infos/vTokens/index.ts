import type { VTokenAssets } from 'libs/tokens/types';
import { ChainId } from 'types';
import { vTokenAssets as arbitrumOneVTokenAssets } from './arbitrumOne';
import { vTokenAssets as arbitrumSepoliaVTokenAssets } from './arbitrumSepolia';
import { vTokenAssets as baseMainnetVTokenAssets } from './baseMainnet';
import { vTokenAssets as baseSepoliaVTokenAssets } from './baseSepolia';
import { vTokenAssets as berachainBepoliaVTokenAssets } from './berachainBepolia';
import { vTokenAssets as berachainMainnetVTokenAssets } from './berachainMainnet';
import { vTokenAssets as bscMainnetVTokenAssets } from './bscMainnet';
import { vTokenAssets as bscTestnetVTokenAssets } from './bscTestnet';
import { vTokenAssets as ethereumVTokenAssets } from './ethereum';
import { vTokenAssets as opBnbMainnetVTokenAssets } from './opBnbMainnet';
import { vTokenAssets as opBnbTestnetVTokenAssets } from './opBnbTestnet';
import { vTokenAssets as optimismMainnetVTokenAssets } from './optimismMainnet';
import { vTokenAssets as optimismSepoliaVTokenAssets } from './optimismSepolia';
import { vTokenAssets as sepoliaVTokenAssets } from './sepolia';
import { vTokenAssets as unichainMainnetVTokenAssets } from './unichainMainnet';
import { vTokenAssets as unichainSepoliaVTokenAssets } from './unichainSepolia';
import { vTokenAssets as zkSyncMainnetVTokenAssets } from './zkSyncMainnet';
import { vTokenAssets as zkSyncSepoliaVTokenAssets } from './zkSyncSepolia';

export const vTokenAssetsPerChainId: Record<ChainId, VTokenAssets> = {
  [ChainId.BSC_MAINNET]: bscMainnetVTokenAssets,
  [ChainId.BSC_TESTNET]: bscTestnetVTokenAssets,
  [ChainId.OPBNB_MAINNET]: opBnbMainnetVTokenAssets,
  [ChainId.OPBNB_TESTNET]: opBnbTestnetVTokenAssets,
  [ChainId.ETHEREUM]: ethereumVTokenAssets,
  [ChainId.SEPOLIA]: sepoliaVTokenAssets,
  [ChainId.ARBITRUM_ONE]: arbitrumOneVTokenAssets,
  [ChainId.ARBITRUM_SEPOLIA]: arbitrumSepoliaVTokenAssets,
  [ChainId.ZKSYNC_MAINNET]: zkSyncMainnetVTokenAssets,
  [ChainId.ZKSYNC_SEPOLIA]: zkSyncSepoliaVTokenAssets,
  [ChainId.OPTIMISM_MAINNET]: optimismMainnetVTokenAssets,
  [ChainId.OPTIMISM_SEPOLIA]: optimismSepoliaVTokenAssets,
  [ChainId.BASE_MAINNET]: baseMainnetVTokenAssets,
  [ChainId.BASE_SEPOLIA]: baseSepoliaVTokenAssets,
  [ChainId.UNICHAIN_MAINNET]: unichainMainnetVTokenAssets,
  [ChainId.UNICHAIN_SEPOLIA]: unichainSepoliaVTokenAssets,
  [ChainId.BERACHAIN_MAINNET]: berachainMainnetVTokenAssets,
  [ChainId.BERACHAIN_BEPOLIA]: berachainBepoliaVTokenAssets,
};
