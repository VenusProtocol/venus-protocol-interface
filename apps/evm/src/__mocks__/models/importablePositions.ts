import { busd, usdc, usdt, weth, xvs } from '__mocks__/models/tokens';
import type { ImportableProtocol, ImportableSupplyPosition } from 'types';

export const importablePositions: {
  [protocol in ImportableProtocol]: ImportableSupplyPosition[];
} = {
  aave: [
    {
      aTokenAddress: `0xfakeA${xvs.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 99.99,
      tokenAddress: xvs.address,
      userATokenBalanceMantissa: 20000000000000000000n,
      userATokenBalanceWithInterestsMantissa: 24000000000000000000n,
      userSupplyBalanceMantissa: 40000000000000000000n,
    },
    {
      aTokenAddress: `0xfakeA${usdc.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.01,
      tokenAddress: usdc.address,
      userATokenBalanceMantissa: 10000000n,
      userATokenBalanceWithInterestsMantissa: 12000000n,
      userSupplyBalanceMantissa: 50000000n,
    },
    {
      aTokenAddress: `0xfakeA${usdt.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.15,
      tokenAddress: usdt.address,
      userATokenBalanceMantissa: 10000000n,
      userATokenBalanceWithInterestsMantissa: 12000000n,
      userSupplyBalanceMantissa: 50000000n,
    },
    {
      aTokenAddress: `0xfakeA${busd.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.02,
      tokenAddress: busd.address,
      userATokenBalanceMantissa: 1n,
      userATokenBalanceWithInterestsMantissa: 1n,
      userSupplyBalanceMantissa: 1n,
    },
    {
      aTokenAddress: `0xfakeA${weth.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.09,
      tokenAddress: weth.address,
      userATokenBalanceMantissa: 180000000000000000000n,
      userATokenBalanceWithInterestsMantissa: 186000000000000000000n,
      userSupplyBalanceMantissa: 200000000000000000000n,
    },
  ],
};
