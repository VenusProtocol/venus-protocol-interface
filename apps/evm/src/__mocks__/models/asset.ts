import BigNumber from 'bignumber.js';

import { busd, usdc, usdt, xvs } from '__mocks__/models/tokens';
import { vBusd, vUsdc, vUsdt, vXvs } from '__mocks__/models/vTokens';

import { Asset } from 'types';

export const assetData: Asset[] = [
  {
    vToken: vXvs,
    supplyApyPercentage: new BigNumber('0.05225450324405023'),
    borrowApyPercentage: new BigNumber('-2.3062487835658776'),
    collateralFactor: 0.5,
    reserveFactor: 0.25,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('127.86734'),
    liquidityCents: new BigNumber(8036465875),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(49.589181233),
    userWalletBalanceTokens: new BigNumber('100'),
    userWalletBalanceCents: new BigNumber('12786'),
    userSupplyBalanceTokens: new BigNumber('90'),
    userSupplyBalanceCents: new BigNumber('11508'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: true,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(70925716),
    supplyBalanceCents: new BigNumber(278311516),
    supplyBalanceTokens: new BigNumber('19339683254955736'),
    borrowBalanceTokens: new BigNumber('1852935.597521220541385584'),
    supplyPercentageRatePerBlock: new BigNumber('0.000001'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000001'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.11720675342484096'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('4.17469243006608279'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
  },
  {
    vToken: vUsdc,
    supplyApyPercentage: new BigNumber('3.887242555711379188'),
    borrowApyPercentage: new BigNumber('-5.361233028654066'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('99.99364'),
    liquidityCents: new BigNumber(1702951959),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1),
    userWalletBalanceTokens: new BigNumber('0'),
    userWalletBalanceCents: new BigNumber('0'),
    userSupplyBalanceTokens: new BigNumber('100'),
    userSupplyBalanceCents: new BigNumber('9999'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(858721657509436),
    supplyBalanceCents: new BigNumber(1000183891879506),
    supplyBalanceTokens: new BigNumber('47171999131879271527200010830'),
    borrowBalanceTokens: new BigNumber('73128318.509651061457900627'),
    supplyPercentageRatePerBlock: new BigNumber('0.000002'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000002'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.353105649796123742'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('0.953105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.753105649796123742'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.670327607690572731'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('1.233105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
  {
    vToken: vUsdt,
    supplyApyPercentage: new BigNumber('3.593608909332766999'),
    borrowApyPercentage: new BigNumber('-4.9748661428011145'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.024602'),
    liquidityCents: new BigNumber(5534102886),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(0.981982),
    userWalletBalanceTokens: new BigNumber('900'),
    userWalletBalanceCents: new BigNumber('90000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('40'),
    userBorrowBalanceCents: new BigNumber('4000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(3158444721),
    supplyBalanceCents: new BigNumber(1098041201011568),
    supplyBalanceTokens: new BigNumber('5029972090817266864401527367893625'),
    borrowBalanceTokens: new BigNumber('232511166.920938849475104194'),
    supplyPercentageRatePerBlock: new BigNumber('0.000003'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000003'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.421719501189155143'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.522209972682294832'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.015310564979612374'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
  },
  {
    vToken: vBusd,
    supplyApyPercentage: new BigNumber('2.886396363044176106'),
    borrowApyPercentage: new BigNumber('-4.050271277344538'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.000922'),
    liquidityCents: new BigNumber(3654492935),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1.000003),
    userWalletBalanceTokens: new BigNumber('110'),
    userWalletBalanceCents: new BigNumber('11000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('50'),
    userBorrowBalanceCents: new BigNumber('5000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(83910350502),
    supplyBalanceCents: new BigNumber(1054707853878),
    supplyBalanceTokens: new BigNumber('51881081291203672464'),
    borrowBalanceTokens: new BigNumber('142662020.229587308931217432'),
    supplyPercentageRatePerBlock: new BigNumber('0.000004'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000004'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.678420831753642169'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: xvs,
        apyPercentage: new BigNumber(0),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.852697602175970714'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: busd,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
];
