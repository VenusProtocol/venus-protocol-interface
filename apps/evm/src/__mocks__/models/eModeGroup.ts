import type { Asset, EModeGroup } from 'types';
import { assetData } from './asset';

const generateEModeGroup = ({
  groupAssets,
  id,
  name,
}: { groupAssets: Asset[]; id: number; name: string }): EModeGroup => ({
  id,
  name,
  assetSettings: groupAssets.map(a => ({
    vToken: a.vToken,
    collateralFactor: a.userCollateralFactor + 0.1,
    liquidationThresholdPercentage: a.userLiquidationThresholdPercentage + 12,
    liquidationPenaltyPercentage: a.userLiquidationThresholdPercentage - 50,
    liquidityCents: a.liquidityCents.toNumber(),
    liquidityTokens: a.cashTokens,
    isBorrowable: a.isBorrowable,
  })),
});

export const eModeGroups: EModeGroup[] = [
  generateEModeGroup({
    id: 0,
    name: 'Stablecoins',
    groupAssets: assetData.slice(0, 3),
  }),
  generateEModeGroup({
    id: 1,
    name: 'DeFi',
    groupAssets: assetData.slice(2, 4),
  }),
  generateEModeGroup({
    id: 2,
    name: '#ToTheMoon',
    groupAssets: assetData.slice(5, 8),
  }),
];
