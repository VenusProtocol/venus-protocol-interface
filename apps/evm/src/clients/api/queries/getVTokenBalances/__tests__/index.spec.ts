import type { PublicClient } from 'viem';

import fakeAccountAddress from '__mocks__/models/address';
import { vBnb, vUsdc, vXvs } from '__mocks__/models/vTokens';

import { getVTokenBalances } from '..';

const vTokens = [vBnb, vXvs, vUsdc];

describe('getVTokenBalances', () => {
  it('returns token balances in the right format on success', async () => {
    const fakePublicClient = {
      multicall: () =>
        vTokens.map(() => ({
          status: 'success',
          result: 100000000000000000n,
        })),
    } as unknown as PublicClient;

    const res = await getVTokenBalances({
      publicClient: fakePublicClient,
      accountAddress: fakeAccountAddress,
      vTokens,
    });

    expect(res).toMatchSnapshot();
  });
});
