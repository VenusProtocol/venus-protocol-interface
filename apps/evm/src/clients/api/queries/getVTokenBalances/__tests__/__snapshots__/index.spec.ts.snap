// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getVTokenBalances > returns token balances in the right format on success 1`] = `
{
  "vTokenBalances": [
    {
      "balanceMantissa": "100000000000000000",
      "vToken": {
        "address": "0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c",
        "decimals": 8,
        "symbol": "vBNB",
        "underlyingToken": {
          "address": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB",
          "asset": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20xmlns:v='https://vecta.io/nano'%3e%3cg%20clip-path='url(%23A)'%3e%3cg%20clip-path='url(%23B)'%3e%3cpath%20fill-rule='evenodd'%20d='M12%200a12%2012%200%201%201%200%2024%2012%2012%200%201%201%200-24z'%20fill='%23f0b90b'/%3e%3cg%20fill='%23fff'%3e%3cpath%20d='M6.595%2012l.009%203.173L9.3%2016.76v1.858l-4.274-2.507v-5.039L6.595%2012zm0-3.173v1.849l-1.57-.929V7.898l1.57-.929%201.578.929-1.578.929zm3.831-.929l1.57-.929%201.578.929-1.578.929-1.57-.929zM7.73%2014.515v-1.858l1.57.929v1.849l-1.57-.92zm2.696%202.91l1.57.929%201.578-.929v1.849l-1.578.929-1.57-.929v-1.849zm5.4-9.527l1.57-.929%201.578.929v1.849l-1.578.929V8.827l-1.57-.929zm1.57%207.275L17.405%2012l1.57-.929v5.038l-4.274%202.507v-1.858l2.695-1.586zm-1.126-.658l-1.57.92v-1.849l1.57-.929v1.858zm0-5.03l.009%201.858-2.704%201.587v3.181l-1.57.92-1.57-.92V12.93l-2.704-1.587V9.485l1.577-.929%202.687%201.594%202.704-1.594%201.578.929h-.007zM7.73%206.313l4.266-2.515%204.274%202.515-1.57.929-2.704-1.594L9.3%207.241l-1.57-.929z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='A'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3cclipPath%20id='B'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e",
          "decimals": 18,
          "isNative": true,
          "symbol": "BNB",
        },
      },
    },
    {
      "balanceMantissa": "100000000000000000",
      "vToken": {
        "address": "0x6d6F697e34145Bb95c54E77482d97cc261Dc237E",
        "decimals": 8,
        "symbol": "vXVS",
        "underlyingToken": {
          "address": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
          "asset": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20xmlns:v='https://vecta.io/nano'%3e%3cg%20clip-path='url(%23B)'%3e%3ccircle%20cx='12'%20cy='12'%20r='12'%20fill='%231f2028'/%3e%3cpath%20d='M18.957%209.442l-5.27%209.128c-.159.275-.388.504-.664.663s-.588.243-.906.243-.631-.084-.906-.243-.505-.387-.664-.663l-.924-1.599c-.003-.019%200-.023.004-.026.019-.003.023%200%20.026.004a1.52%201.52%200%200%200%201.267.425%201.52%201.52%200%200%200%20.642-.241c.193-.128.354-.299.47-.499l4.412-7.649a1.52%201.52%200%200%200%20.086-1.341%201.52%201.52%200%200%200-1.012-.884c-.017-.009-.019-.014-.019-.019.009-.017.014-.019.019-.019h1.87c.318%200%20.631.084.906.244s.504.388.663.664.243.588.242.907-.084.631-.243.907zm-6.239-2.721h-1.827c-.012.01-.013.014-.013.018.006.014.009.017.013.018.123.048.234.121.326.216s.163.208.207.332.061.257.049.388-.051.259-.117.373l-2.663%204.606a.92.92%200%200%201-.648.441.92.92%200%200%201-.752-.22c-.017-.008-.022-.007-.026-.004-.008.017-.007.022-.004.026l.935%201.623c.105.182.256.333.438.439s.389.161.599.161.417-.055.599-.161.333-.257.438-.439l3.484-6.022c.105-.182.16-.389.16-.599s-.055-.417-.161-.599-.257-.333-.439-.438-.389-.16-.599-.16zm-6.342%200c-.312%200-.617.092-.876.266s-.461.419-.581.708-.151.605-.09.911.211.587.431.807.501.371.807.431.623.03.911-.09.534-.321.708-.581.266-.564.266-.876c0-.207-.04-.412-.12-.604s-.195-.365-.342-.512-.32-.263-.512-.342-.396-.12-.604-.12z'%20fill='url(%23A)'/%3e%3c/g%3e%3cdefs%3e%3clinearGradient%20id='A'%20x1='19.016'%20y1='16.814'%20x2='2.55'%20y2='5.633'%20gradientUnits='userSpaceOnUse'%3e%3cstop%20stop-color='%235433ff'/%3e%3cstop%20offset='.5'%20stop-color='%2320bdff'/%3e%3cstop%20offset='1'%20stop-color='%235cffa2'/%3e%3c/linearGradient%3e%3cclipPath%20id='B'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e",
          "decimals": 18,
          "symbol": "XVS",
        },
      },
    },
    {
      "balanceMantissa": "100000000000000000",
      "vToken": {
        "address": "0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7",
        "decimals": 8,
        "symbol": "vUSDC",
        "underlyingToken": {
          "address": "0x16227D60f7a0e586C66B005219dfc887D13C9531",
          "asset": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20xmlns:v='https://vecta.io/nano'%3e%3cg%20clip-path='url(%23A)'%3e%3cpath%20d='M12%2024a11.97%2011.97%200%200%200%2012-12A11.97%2011.97%200%200%200%2012%200%2011.97%2011.97%200%200%200%200%2012a11.97%2011.97%200%200%200%2012%2012z'%20fill='%232775ca'/%3e%3cg%20fill='%23fff'%3e%3cpath%20d='M15.3%2013.9c0-1.75-1.05-2.35-3.15-2.6-1.5-.2-1.8-.6-1.8-1.3s.5-1.15%201.5-1.15c.9%200%201.4.3%201.65%201.05.05.15.2.25.35.25h.8A.34.34%200%200%200%2015%209.8v-.05c-.2-1.1-1.1-1.95-2.25-2.05V6.5c0-.2-.15-.35-.4-.4h-.75c-.2%200-.35.15-.4.4v1.15c-1.5.2-2.45%201.2-2.45%202.45%200%201.65%201%202.3%203.1%202.55%201.4.25%201.85.55%201.85%201.35s-.7%201.35-1.65%201.35c-1.3%200-1.75-.55-1.9-1.3-.05-.2-.2-.3-.35-.3h-.85a.34.34%200%200%200-.35.35v.05c.2%201.25%201%202.15%202.65%202.4v1.2c0%20.2.15.35.4.4h.75c.2%200%20.35-.15.4-.4v-1.2c1.5-.25%202.5-1.3%202.5-2.65zm-5.85%205.25A7.45%207.45%200%200%201%205%209.55c.75-2.1%202.4-3.7%204.45-4.45a.5.5%200%200%200%20.3-.5v-.7c0-.2-.1-.35-.3-.4-.05%200-.15%200-.2.05C4.5%205.05%201.9%2010.1%203.4%2014.85c.9%202.8%203.05%204.95%205.85%205.85.2.1.4%200%20.45-.2.05-.05.05-.1.05-.2v-.7c0-.15-.15-.35-.3-.45zm5.3-15.6c-.2-.1-.4%200-.45.2-.05.05-.05.1-.05.2v.7c0%20.2.15.4.3.5a7.45%207.45%200%200%201%204.45%209.6c-.75%202.1-2.4%203.7-4.45%204.45a.5.5%200%200%200-.3.5v.7c0%20.2.1.35.3.4.05%200%20.15%200%20.2-.05%204.75-1.5%207.35-6.55%205.85-11.3-.9-2.85-3.1-5-5.85-5.9z'/%3e%3c/g%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='A'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e",
          "decimals": 6,
          "symbol": "USDC",
        },
      },
    },
  ],
}
`;
